### PR TITLE
fix: Fix recurring booking failure due to rate limiting

### LINKED ISSUE #10673: [CAL-2517] Recurring Event - Error while booking a meeting.
### Issue Summary
I get the Couldn't Book Meeting Error while trying to book a recurring meeting. 

### Steps to Reproduce

1. Book a recurring event type

Screenshot

[Chrome logs](https://paste.c-net.org/FelicityComrades)

<sub>[CAL-2517](https://linear.app/calcom/issue/CAL-2517/error-while-booking-a-meeting)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

"core" has a limit of 10 requests every 60s. Due to `handleNewBooking` called from `recurring-event` handler multiple times depending on the recurrence, it can fail.

Moved the rateLimiting logic to **actual route** to guarantee that it's called once per route.

Note: I think we should have some eslint rule to disallow rateLimiting in non api route files because it can accidentally cause issues. cc @sean-brydon @keithwillcode 

image

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- Create a stash account and setup env variables
- Reduce the "core" limit here to a small number say 2(https://github.com/calcom/cal.com/blob/15734921d34dec3af2b5a37ffa0e95fbde680541/packages/lib/rateLimit.ts#L58)
- Try booking a recurring event in main and it will fail(num intervals should alteast be 3)
- In this branch, it will pass
- Also, test that if in less than a minute 3 normals bookings are done. Third one fails.

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist
- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12322** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12322?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12322?utm_source=stack-comment).

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 1]
```diff
@@ -630,12 +628,6 @@ async function handler(
   }
 ) {
   const { userId } = req;
-  const userIp = getIP(req);
-
-  await checkRateLimitAndThrowError({
```
Comment (hariombalhara) [inline packages/features/bookings/lib/handleNewBooking.ts:635]: We shouldn't call rate limit here as this function is called by recurring-event route many times.

File: apps/api/pages/api/bookings/_post.ts [hunk 2]
```diff
@@ -205,6 +207,13 @@ import { defaultResponder } from "@calcom/lib/server";
 async function handler(req: NextApiRequest) {
   const { userId, isAdmin } = req;
   if (isAdmin) req.userId = req.body.userId || userId;
+  const userIp = getIP(req);
+
+  await checkRateLimitAndThrowError({
```
Comment (keithwillcode) [inline apps/api/pages/api/bookings/_post.ts:212]: Why is this using "core" instead of "api"?

Comment (keithwillcode) [reply apps/api/pages/api/bookings/_post.ts:212]: We already have a global rate limit check by the API key

**Review by keithwillcode — CHANGES_REQUESTED** [review]
Requesting changes for clarification on API route rate limiting

File: apps/api/pages/api/bookings/_post.ts (hunk 2 above)
Comment (hariombalhara) [reply apps/api/pages/api/bookings/_post.ts:212]: `handleNewBooking` which was called from here was using core only. This is the reason.

Comment (hariombalhara) [reply apps/api/pages/api/bookings/_post.ts:212]: I think it makes sense to remove it then.

It also means that there were two different rate limitings in place for this route

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**